File: python/potega.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  potega.py

def potega_it(a, n):
    
    wynik = 1
    
    for i in range(n):
        
        wynik = wynik * a
        #print(wynik)
     
    return wynik

def main(args):
    
    #a = int(input("Podaj podstawę: "))
    #n = int(input("Podaj wykładnik: "))
    
    #print("Potega {} do {} wynosi {}".format(a, n, potega_it(a, n)))
    assert(potega_it(3 ,2) == 9)
    assert(potega_it(3, 3) == 27)
    assert(potega_it(0, 2) == 0)
    
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/potega.cpp
/*
 * potega.cpp
 */


#include <iostream>
using namespace std;

int potega_it(int a, int n) {
    int wynik = 1;
    for (int i = 1; i <= n; i++) {
        wynik = wynik * a;
    }
    return wynik;
}
// 2^3 = 2^2 * 2
//a^n = a(n-1) * a
int potega_re(int a, int n) {
    if (n == 0) return 1;
    return potega_re(a, n - 1) * a;
    
    
        
}

int main(int argc, char **argv)
{
	int a, n;
    cout << "Podaj podstawe i wykladnik: ";
    cin >> a >> n;
    potega_it(a, n);
    cout << "Potega: " << potega_it(a, n) << endl;
    cout << "Potega: " << potega_re(a, n) << endl;
    
    
    
	return 0;
}

